### PR TITLE
Get working on 16bpp RGB565 framebuffer device.

### PR DESCRIPTION
**Pull request by dlech**

### CODE REVIEW

**Comment by dlech** [comment]
See #25

File: ev3dev.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 import fcntl
 import array
 import mmap
+import struct
 import ctypes
 from PIL import Image, ImageDraw
 from struct import unpack
```
Comment (dlech) [inline ev3dev.py]: I see now that I should have added pack here.

File: ev3dev.py [hunk 2]
```diff
@@ -2262,11 +2263,24 @@ def clear(self):
...
 
+    def _color565(self, r, g, b):
+	"""Convert red, green, blue components to a 16-bit 565 RGB value. Components
+	should be values 0 to 255.
+	"""
+	return (((r & 0xF8) << 8) | ((g & 0xFC) << 3) | (b >> 3))
+
+    def _img_to_rgb565_bytes(self):
+        pixels = [self._color565(r,g,b) for (r,g,b) in self._img.getdata()]
+        return struct.pack('H' * len(pixels), *pixels)
```
Comment (dlech) [inline ev3dev.py]: Don't know if this is the most efficient way to do this, but it works.

Comment (rhempel) [inline ev3dev.py]: Make it work, then make it fast!

Comment (ddemidov) [inline ev3dev.py]: The most efficient way would probably be using numpy, but I am not sure we want that as dependency. May be array would also work, I can look at it later.

Comment (ddemidov) [inline ev3dev.py]: Ok, it seems python array is just an efficient way to *store* data arrays; it does not provide any array manipulation functions. However, I measured performance of @dlech's list-based solution and PiStorms numpy-based one. numpy is about 64 times faster (on my desktop computer): http://nbviewer.ipython.org/gist/ddemidov/84053a30e443645fcbeb

**Comment by rhempel** [comment]
Thanks!

**Pull request merged**